tests/integration/staking/keeper/validator_bench_test.go

```diff
@@ -27,3 +36,121 @@ func BenchmarkGetValidator(b *testing.B) {
 		}
 	}
 }
+
+func BenchmarkGetValidatorDelegations(b *testing.B) {
+	var totalPower int64
+	powersNumber := 10
+
+	powers := make([]int64, powersNumber)
+	for i := range powers {
+		powers[i] = int64(i)
+		totalPower += int64(i)
+	}
+
+	app, ctx, _, valAddrs, vals := initValidators(b, totalPower, len(powers), powers)
+	for _, validator := range vals {
+		app.StakingKeeper.SetValidator(ctx, validator)
+	}
+
+	delegationsNum := 1000
+	delegators := make([]sdk.AccAddress, delegationsNum)
+	for _, val := range valAddrs {
+		for i := 0; i < delegationsNum; i++ {
+			delegator := sdk.AccAddress(fmt.Sprintf("address%d", i))
+			banktestutil.FundAccount(app.BankKeeper, ctx, delegator,
+				sdk.NewCoins(sdk.NewCoin(sdk.DefaultBondDenom, sdk.NewInt(int64(i)))))
+			NewDel := types.NewDelegation(delegator, val, sdk.NewDec(int64(i)))
+			app.StakingKeeper.SetDelegation(ctx, NewDel)
+			delegators[i] = delegator
```

Comment:
Aren't you overwritting the slice since you're in a loop over valAddrs?

Reply:
yeah, this is being overwritten. BTW we are not using this array anywhere further and removed it.